mage_integrations/mage_integrations/destinations/redshift/__init__.py

```diff
@@ -132,31 +142,40 @@ def build_insert_commands(
 
         if unique_constraints and UNIQUE_CONFLICT_METHOD_UPDATE == unique_conflict_method:
             full_table_name_temp = self.full_table_name(schema_name, table_name, prefix='temp_')
-            full_table_name_old = self.full_table_name(schema_name, table_name, prefix='old_')
             drop_temp_table_command = f'DROP TABLE IF EXISTS {full_table_name_temp}'
-            drop_old_table_command = f'DROP TABLE IF EXISTS {full_table_name_old}'
             unique_constraints_clean = [
                 f'{self.clean_column_name(col)}'
                 for col in unique_constraints
             ]
+
+            drop_duplicate_records_from_temp = (f'DELETE FROM {full_table_name_temp} '
+            f'WHERE ({", ".join(unique_constraints_clean)},_mage_created_at) '
+            f'IN ( SELECT {", ".join(unique_constraints_clean)}, _mage_created_at '
+            f'FROM ( SELECT {", ".join(unique_constraints_clean)}, _mage_created_at, ROW_NUMBER() '
+            f'OVER (PARTITION BY {", ".join(unique_constraints_clean)} '
+            f'ORDER BY _mage_created_at DESC) as row_num '
+            f'FROM {full_table_name_temp} ) AS subquery_alias WHERE row_num > 1);')
```

Comment:
can you share an example of what this query looks like with the variable values filled in?

Reply:
@dy46 Please find sample query here

DELETE FROM temp_abc
            WHERE (id,_mage_created_at)
            IN ( SELECT id, _mage_created_at
            FROM ( SELECT id, _mage_created_at, ROW_NUMBER()
            OVER (PARTITION BY id
            ORDER BY _mage_created_at DESC) as row_num
            FROM temp_abc ) AS subquery_alias WHERE row_num > 1);

Reply:
@dy46 changes done as suggested